### PR TITLE
feat: head tag supports extension for template

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind feature
/milestone 2.0
/area core

#### What this PR does / why we need it:
允许插件通过实现 TemplateHeadProcessor 接口来修改主题模板的 head 标签

#### Which issue(s) this PR fixes:
how to test it?
1. 克隆 https://github.com/halo-sigs/plugin-umami
2. build 一个 jar 包作为插件使用
3. 配置 plugin-umami 后能在主题页的 head 标签看到一个用于 umami 统计的 script 标签

Fixes #

#### Special notes for your reviewer:
/cc @halo-dev/sig-halo 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by ruibaby — COMMENTED** [review]
/lgtm

image

image

File: src/main/java/run/halo/app/theme/dialect/GlobalHeadInjectionProcessor.java [hunk 1]
```diff
@@ -75,8 +74,14 @@ protected void doProcess(ITemplateContext context, IModel model,
...
-    private Collection<TemplateHeadProcessor> getTemplateHeadProcessors(ApplicationContext ctx) {
-        return ctx.getBeansOfType(TemplateHeadProcessor.class)
-            .values();
+    private Collection<TemplateHeadProcessor> getTemplateHeadProcessors(ITemplateContext context) {
+        final ApplicationContext appCtx = SpringContextUtils.getApplicationContext(context);
+        try {
+            ExtensionComponentsFinder componentsFinder =
+                appCtx.getBean(ExtensionComponentsFinder.class);
+            return componentsFinder.getExtensions(TemplateHeadProcessor.class);
+        } catch (NoSuchBeanDefinitionException e) {
```
Comment (JohnNiang) [inline src/main/java/run/halo/app/theme/dialect/GlobalHeadInjectionProcessor.java:83]: 建议移除没有必要的 catch。ExtensionComponentsFinder 已经包含 `@Component` 注解。

**Review by JohnNiang — CHANGES_REQUESTED** [review]

**Review by JohnNiang — APPROVED** [review]
/approve
/lgtm

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Pull request merged**